### PR TITLE
feat: add offline docs

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Related to https://github.com/coder/coder/issues/5466

Screen Shot 2023-07-14 at 15 30 36

### CODE REVIEW

**Comment by bpmct** [comment]
So this is going to be in a different artifact? https://github.com/coder/coder/discussions/8110

**Comment by BrunoQuaresma** [comment]
@bpmct Yes, so the user can download the .tar file and host the docs where they want to.

**Comment by matifali** [comment]
@BrunoQuaresma but we should make it linkable from the Coder dashboard.

**Comment by BrunoQuaresma** [comment]
You can follow the RFC discussion to see the reasons we decided to not do it rn but it can be added in the future if we see more users asking for the feature.

**Comment by bpmct** [comment]
Ah, but I thought we are going to support setting a custom docs link URL for in-product docs links?

`CODER_DOCS_URL=https://coder-docs.example.com/`

Screenshot 2023-07-15 at 2 56 10 PM

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -590,6 +590,31 @@ jobs:
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.20.1
+      - run: yarn install
+        working-directory: ./offlinedocs
+      - run: yarn export
+        working-directory: ./offlinedocs
+      - name: Tar files
+        run: tar -cvf docs.tar offlinedocs/out
+      - name: Upload Artifact
```
Comment (matifali) [inline .github/workflows/ci.yaml:612]:
```suggestion
      - name: Upload Artifact
      	if: github.ref == 'refs/heads/main'
```
If you add this then we only upload artifacts when PR is merged with the `main` branch.

**Review by matifali — COMMENTED** [review]
Also, you may wish to keep the dependencies updated by adding an entry for offlinedocs in `.github/dependabot.yaml`

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (matifali) [reply .github/workflows/ci.yaml:612]: Also, we may remove it entirely from the ci.yaml and include it in release.yaml. 
It's better if we only build on new releases.

**Comment by matifali** [comment]
I like the light theme.

**Comment by BrunoQuaresma** [comment]
> Ah, but I thought we are going to support setting a custom docs link URL for in-product docs links?
> 
> `CODER_DOCS_URL=https://coder-docs.example.com/`
> 
> Screenshot 2023-07-15 at 2 56 10 PM

We definitely can do it. It is very easy, thanks for bringing this.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (BrunoQuaresma) [reply .github/workflows/ci.yaml:612]: Yes, 100%. The idea to make it on the PR, for now, was to test it before adding it to the release file.

File: offlinedocs/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { ChakraProvider, extendTheme } from "@chakra-ui/react"
```
Comment (deansheather) [inline offlinedocs/pages/_app.tsx:1]: We usually use 2 space indentation for ts and tsx files, this appears to be using tabs.

The other files in this PR seem to also be using tabs

File: offlinedocs/scripts/copyImages.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+cp ../docs/images public/images --recursive
```
Comment (deansheather) [inline offlinedocs/scripts/copyImages.sh:3]: Should `rm` first so any deleted images get deleted from here as well

Comment (BrunoQuaresma) [reply offlinedocs/scripts/copyImages.sh:3]: Do you mean something like `rm -rf public/images && cp ../docs/images public/images --recursive`?

File: offlinedocs/pages/_app.tsx (hunk 2 above)
Comment (BrunoQuaresma) [reply offlinedocs/pages/_app.tsx:1]: Good catch, I will fix that.

**Review by matifali — APPROVED** [review]
I would suggest adding an entry for offline docs to `dependabot.yaml`.
Other than that ci looks good.

**Comment by BrunoQuaresma** [comment]
I think for now it is ok to not have a dependabot for the offlinedocs but I will keep that in mind for sure.

File: offlinedocs/pages/docs/[[...slug]].tsx [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+            <SidebarNav nav={nav} version={version} border={0} />
+          </DrawerBody>
+        </DrawerContent>
+      </Drawer>
+    </>
+  )
+}
+
+const slugifyTitle = (title: string) => {
+  return _.kebabCase(title.toLowerCase())
```
Comment (deansheather) [inline offlinedocs/pages/docs/[[...slug]].tsx:355]: IMO we should avoid the lodash dep if we're only using it for this function, you could inline the function here instead perhaps

File: offlinedocs/scripts/copyImages.sh (hunk 3 above)
Comment (deansheather) [reply offlinedocs/scripts/copyImages.sh:3]: Yeah, but it can go on separate lines instead of using `&&` since it's a shell file

**Review by deansheather — APPROVED** [review]

**Comment by deansheather** [comment]
I changed it to `.tgz` (aka. `.tar.gz`) to match our Helm artifact. It'll show up in the release as `coder_docs_$(VERSION).tgz`

File: offlinedocs/pages/docs/[[...slug]].tsx (hunk 4 above)
Comment (BrunoQuaresma) [reply offlinedocs/pages/docs/[[...slug]].tsx]: it is already been used for other packages tho, so no big deal IMO.

**Pull request merged**